### PR TITLE
Add macro to build params

### LINKED ISSUE #275: [jsonrpsee types]: provide macro for constructing JsonRpcParams
The `JsonRpcParams` are a little boiler-platy to provide as we require the arguments to be `serde_json::Value`, 

Example:
```rust
// MyType doesn't implement `Into<serde_json::Value>`
let x: = MyType { };
let inner: &[JsonValue] = &[0_u32.into(), "hoo".to_string().into(), serde_json::to_value(x).unwrap()];
let params =  JsonRpcParams::ArrayRef(inner);
```

So it would be nice is to provide a macro such as `params_positional!` and `params_object` to avoid that.
Perhaps just provide a wrapper on-top of [serde_json::json!](https://docs.serde.rs/serde_json/macro.json.html)?

Note: Copy @kianenigma's stuff.

### PR DESCRIPTION
**Pull request by dvdplm**
Add a convenience macro to build `ParamsSer` for use in clients. Ty @kianenigma.

Closes #275

### CODE REVIEW

**Review by maciejhirsz — APPROVED** [review]
Looks clean!

File: examples/ws_subscription.rs [hunk 1]
```diff
@@ -25,28 +25,28 @@
...
 	ws_client::WsClientBuilder,
 	ws_server::{RpcModule, WsServerBuilder},
 };
 use std::net::SocketAddr;
 
-const NUM_SUBSCRIPTION_RESPONSES: usize = 10;
-
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
+	const NUM_SUBSCRIPTION_RESPONSES: usize = 5;
```
Comment (insipx) [inline examples/ws_subscription.rs:37]: why move const into the scope of a function, instead of just using a let/5 in the loop

**Review by insipx — APPROVED** [review]
lgtm other than minor nitpick

File: examples/ws_subscription.rs (hunk 1 above)
Comment (dvdplm) [reply examples/ws_subscription.rs:37]: You're right, this is dumb. Will fix.

File: utils/src/client.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+//! Shared utilities for `jsonrpsee` clients.
+
+#[doc(hidden)]
+pub mod __reexports {
+	pub use jsonrpsee_types::{to_json_value, v2::ParamsSer};
+}
+
+#[macro_export]
+/// Convert the given values to a [`ParamsSer`] as expected by a jsonrpsee Client (http or websocket).
```
Comment (niklasad1) [inline utils/src/client.rs:35]: I guess we could document that this panics if the serialization fails which is very unlikely but possible I guess....

**Pull request merged**